Assets/Scripts/Goal.cs

```diff
@@ -38,18 +43,28 @@ public class Goal : MonoBehaviour {
         }
     }
 
+    public void ResetGoal() {
+        InitRotationAndOffset();
+        goalState = GoalState.FOLLOWING;
+    }
+
     // Update is called once per frame
     void Update() {
         HandleGoalPosition();
     }
     
     private void HandleGoalPosition() {
-        parentPos = transform.parent.position;
-        // Prevent goal from exceeding room bounds
-        newPos.x = Mathf.Clamp(parentPos.x + X_OFFSET, X_MIN, X_MAX);
-        newPos.y = Mathf.Clamp(parentPos.y + Y_OFFSET, Y_MIN, Y_MAX);
-        newPos.z = Mathf.Clamp(parentPos.z + zOffset, Z_MIN, Z_MAX);
-        UpdateGoalPosition(newPos);
+        if (goalState == GoalState.FOLLOWING) {
+            parentPos = transform.parent.position;
+            // Prevent goal from exceeding room bounds
+            newPos.x = Mathf.Clamp(parentPos.x + X_OFFSET, X_MIN, X_MAX);
+            newPos.y = Mathf.Clamp(parentPos.y + Y_OFFSET, Y_MIN, Y_MAX);
+            newPos.z = Mathf.Clamp(parentPos.z + zOffset, Z_MIN, Z_MAX);
```

Comment:
Switching from stationary to following causes it to teleport, which may not be desirable. Try
```
            newPos.x = Mathf.Clamp(Mathf.Lerp(lastSafePos.x, parentPos.x + X_OFFSET, Time.deltaTime), X_MIN, X_MAX);
            newPos.y = Mathf.Clamp(Mathf.Lerp(lastSafePos.y, parentPos.y + Y_OFFSET, Time.deltaTime), Y_MIN, Y_MAX);
            newPos.z = Mathf.Clamp(Mathf.Lerp(lastSafePos.z, parentPos.z + zOffset, Time.deltaTime), Z_MIN, Z_MAX);
``` 
or adding a state for transiting from Stationary to Following

Reply:
is that not supposed to happen?

Reply:
personally i feel it's not very good UX for the goal to be teleporting but if it's expected behaviour for you then